lib/spack/spack/compilers/clang.py

```diff
@@ -42,18 +42,16 @@ class Clang(Compiler):
     cxx_names = ['clang++']
 
     # Subclasses use possible names of Fortran 77 compiler
-    f77_names = ['gfortran']
+    f77_names = ['flang']
 
     # Subclasses use possible names of Fortran 90 compiler
-    fc_names = ['gfortran']
+    fc_names = ['flang']
 
     # Named wrapper links within spack.build_env_path
     link_paths = {'cc': 'clang/clang',
                   'cxx': 'clang/clang++',
-                  # Use default wrappers for fortran, in case provided in
-                  # compilers.yaml
-                  'f77': 'clang/gfortran',
```

Comment:
Regardless of auto detection, unfortunately this likely breaks usage of clang+gfortran as some packages like openblas need compiler wrappers to be named correctly.

Reply:
if only we could make wrapper dynamic, namely, based on what was found as a Fortran compiler, they would point to a dummy `clang/gfortran` or `clang/flang`.

Reply:
Should we do something about that in this PR?

Reply:
Yes, otherwise it looks like you break Spack for all macOS users. Can u add if/else to at least have different difinitiins of link_paths when on macOS and otherwise.

Reply:
Ok, will do? `flang` works on OSX as well, but let's add that later.